transport/http/transport.go

```diff
@@ -147,7 +190,8 @@ func (a *Transport) Start() error {
 // Stop stops the HTTP transport.
 func (a *Transport) Stop() error {
 	return a.once.Stop(func() error {
-		return nil // Nothing to do
+		a.connectorsGroup.Wait()
```

Comment:
maybe for now we should just have a boolean check here to see if there are any peers left in the "peers" map, and if there are, we return an error.